extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/telemetry/SendingInterceptor.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkus.websockets.next.runtime.telemetry;
+
+import io.vertx.core.buffer.Buffer;
+
+/**
+ * Intercepts text or binary send from the {@link io.quarkus.websockets.next.runtime.WebSocketConnectionBase} connection.
+ */
+public sealed interface SendingInterceptor permits MicrometerSendingInterceptor, SendingInterceptor.Noop {
+
+    // following methods should mirror WebSocketConnectionBase sending methods
+
+    void onSend(String text);
```

Comment:
Why this method returns `void` and the `Buffer` version returns `Runnable`?

Reply:
at one point, I used this method also elsewhere, which didn't require runnable. right now, `Runnable` can be used here as well and avoid lambda used together with this method